iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/PmcXmlHandler.java

```diff
@@ -126,36 +125,32 @@ public void startElement(String uri, String localName, String qName,
 			if (articleType!=null) {
 				builder.setEntityType(articleType);	
 			}
-		} else if (hasAmongParents(qName, ELEM_JOURNAL_TITLE, this.parents, ELEM_JOURNAL_META)) {
-			this.currentValue = new StringBuilder();
-		} else if (isWithinElement(qName, ELEM_ARTICLE_ID, ELEM_ARTICLE_META)) {
-			this.currentArticleIdType = attributes.getValue(PUB_ID_TYPE);
-			this.currentValue = new StringBuilder();
 		} else if (isWithinElement(qName, ELEM_FPAGE, ELEM_ARTICLE_META) ||
-				isWithinElement(qName, ELEM_LPAGE, ELEM_ARTICLE_META)) {
-			this.currentValue = new StringBuilder();
-		} else if (hasAmongParents(qName, ELEM_AFFILIATION, this.parents, ELEM_ARTICLE_META)) {
-			this.currentValue = new StringBuilder();
-		} else if (hasAmongParents(qName, ELEM_ARTICLE_TITLE, this.parents, ELEM_REF, ELEM_REF_LIST) ||
+				isWithinElement(qName, ELEM_LPAGE, ELEM_ARTICLE_META) ||
+				isWithinElement(qName, ELEM_SURNAME, ELEM_NAME) ||
+				isWithinElement(qName, ELEM_GIVEN_NAMES, ELEM_NAME) ||
+				hasAmongParents(qName, ELEM_JOURNAL_TITLE, this.parents, ELEM_JOURNAL_META) ||
+				hasAmongParents(qName, ELEM_AFFILIATION, this.parents, ELEM_ARTICLE_META) ||
+				hasAmongParents(qName, ELEM_ARTICLE_TITLE, this.parents, ELEM_REF, ELEM_REF_LIST) ||
 				hasAmongParents(qName, ELEM_SOURCE, this.parents, ELEM_REF, ELEM_REF_LIST) ||
 				hasAmongParents(qName, ELEM_YEAR, this.parents, ELEM_REF, ELEM_REF_LIST) ||
 				hasAmongParents(qName, ELEM_VOLUME, this.parents, ELEM_REF, ELEM_REF_LIST) ||
 				hasAmongParents(qName, ELEM_ISSUE, this.parents, ELEM_REF, ELEM_REF_LIST) ||
 				hasAmongParents(qName, ELEM_FPAGE, this.parents, ELEM_REF, ELEM_REF_LIST) ||
 				hasAmongParents(qName, ELEM_LPAGE, this.parents, ELEM_REF, ELEM_REF_LIST)) {
-			this.currentValue = new StringBuilder();
-		} else if (isWithinElement(qName, ELEM_SURNAME, ELEM_NAME) ||
-				isWithinElement(qName, ELEM_GIVEN_NAMES, ELEM_NAME)) {
-			this.currentValue = new StringBuilder();
+			this.currentValue.delete(0, currentValue.length());
+		} else if (isWithinElement(qName, ELEM_ARTICLE_ID, ELEM_ARTICLE_META)) {
+			this.currentArticleIdType = attributes.getValue(PUB_ID_TYPE);
+			this.currentValue.delete(0, currentValue.length());
```

Comment:
True. Just checked the logic behind both methods and it seems `StringBuffer.setLength(0)` also outperforms `StringBuffer#delete()` method.